src/Domain/ElementDistribution.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <utility>
+#include <vector>
+
+#include "Domain/Structure/ElementId.hpp"
+
+namespace domain {
+
+/*!
+ * \brief Distribution strategy based on grouping elements within blocks
+ * together, and using a 'Z-curve' to determine distribution within blocks.
+ *
+ * \details This class suggests starting placements of elements on processors by
+ * first greedily assigning portions of blocks to each processor, then dividing
```

Comment:
I'll add better details to the documentation, but I'll also answer the questions here:
- when blocks < procs it will still operate on greedy assignment, it's just a little less obvious what happens with the greedy algorithm. Say we have 2 blocks and 5 procs, then it assigns the first 2/5 of a block to the first proc, then then next 2/5 to the next proc, and the third proc gets split between the blocks, etc.
- I currently don't handle multiple nodes differently from multiple cores, though I do see how a more aggressive hierarchy could be useful there - trying to minimize communication across nodes by potentially increasing communication among cores. Here, this just generally tries to evenly minimize communication across the board which naturaally contains the internode communication but doesn't prioritize it.

Thanks for the suggested doc update, I'll incorporate that with other expansions.

Reply:
Thanks! The new docs is a big improvement :)